torch/csrc/jit/graph_executor.cpp

```diff
@@ -445,11 +445,11 @@ struct GraphExecutorImpl {
   void runOptimization(std::shared_ptr<Graph>& graph, const ArgumentSpec& spec) {
     EliminateDeadCode(graph);
     EliminateCommonSubexpression(graph);
-    ConstantPooling(graph);
     UnrollLoops(graph);
     PeepholeOptimize(graph);
     CheckInplace(graph);
     BatchMM(graph);
+    ConstantPooling(graph);
```

Comment:
You raised a valid point when you said that constants will count towards the size of loop in loop unrolling, and so we should use that before that happens. It should be ok to call this pass twice.

Reply:
Oh I see you changed loop unrolling. Never mind my comment.